android/app/src/main/java/org/uwaterloo/subletr/pages/chat/ChatListingPageView.kt

```diff
@@ -0,0 +1,205 @@
+		}
+	} else if (uiState is ChatListingPageUiState.Loaded) {
+		LazyColumn(modifier = modifier) {
+			item {
+				Box(
+					modifier = Modifier
+						.fillMaxWidth()
+						.height(dimensionResource(id = R.dimen.xxl))
+						.padding(horizontal = dimensionResource(id = R.dimen.s))
+						.background(Color.Transparent),
+					contentAlignment = Alignment.BottomStart,
+				) {
+					Text(
+						stringResource(id = R.string.messages),
+						style = MaterialTheme.typography.titleMedium,
+					)
+				}
+			}
+			item {
+				DisplayEntries(uiState.contacts)
+			}
+		}
+		if (uiState.contacts.isEmpty()) {
+			Column(
+				modifier = modifier
+					.fillMaxSize()
+					.background(Color.Transparent),
+				horizontalAlignment = Alignment.CenterHorizontally,
+				verticalArrangement = Arrangement.Center,
+			) {
+				Text(
+					stringResource(id = R.string.empty_chat),
+					color = primaryTextColor,
+				)
+			}
+		}
+	}
+}
+
+@Composable
+fun ListItem(name : String, lastMsg : String) {
+	Box(
+		modifier = Modifier
+			.padding(
+				vertical = dimensionResource(id = R.dimen.xs),
+				horizontal = dimensionResource(id = R.dimen.s),
+			)
+			.wrapContentHeight()
+			.fillMaxWidth(1.0f)
+			.clip(
+				RoundedCornerShape(
+					dimensionResource(id = R.dimen.s)
+				)
+			)
+			.background(secondaryButtonBackgroundColor),
+		) {
+			Row(verticalAlignment = Alignment.CenterVertically) {
+				Icon(name = name)
+				Column(
+					modifier = Modifier
+						.weight(1f)
+						.padding(vertical = dimensionResource(id = R.dimen.s)),
+					verticalArrangement = Arrangement.Center,
+				)
+
+				{
+					Text(text = name,
+						style = MaterialTheme.typography.titleSmall)
+					Text(text = lastMsg,
+						style = MaterialTheme.typography.bodySmall)
+				}
+				Notification()
+			}
+
+	}
+}
+
+@Composable
+fun Icon(name : String) {
+	Box(
+		modifier = Modifier
+			.padding(dimensionResource(id = R.dimen.s))
+			.clip(CircleShape),
+	) {
+		Box(
+			modifier = Modifier
+				.width(dimensionResource(id = R.dimen.xxl))
+				.height(dimensionResource(id = R.dimen.xxl))
+				.background(subletrPink),
+			contentAlignment = Alignment.Center,
+		) {
+			Text(
+				text = name.first().toString(),
+				style = MaterialTheme.typography.titleSmall,
+				color = textOnSubletrPink,
+			)
+		}
+	}
+}
+
+@Composable
+fun Notification(n : Int = 1) {
+	Box(
+		modifier = Modifier
+			.padding(dimensionResource(id = R.dimen.s))
+			.clip(CircleShape),
+	) {
+		Box(
+			modifier = Modifier
+				.width(dimensionResource(id = R.dimen.l))
+				.height(dimensionResource(id = R.dimen.l))
+				.background(Color(0xFF6ABC95)),
+			contentAlignment = Alignment.Center,
+		) {
+			Text(
+				text = n.toString(),
+				color = textOnSubletrPink,
+				style = MaterialTheme.typography.labelLarge,
+			)
+		}
+	}
+}
+
+@Composable
+fun DisplayEntries(contacts : List<Contact>) {
+	(contacts.indices).forEach { i ->
+		ListItem(name = contacts[i].name, lastMsg = contacts[i].msg.last())
+	}
```

Comment:
This is really weird, why are we getting the indices to then call forEach and access the elements by index?
This should just be
`contacts.forEach { contact -> ... }`

Reply:
done